Expression splitter fails on thetis test

Clone thetis from thetisproject/thetis

Run 

```
py.test "test/swe2d/test_steady_state_channel_mms.py::test_steady_state_channel_mms[pndp(n+1)]"
```
```
../../src/deps/COFFEE/coffee/plan.py:110: in _generate_cpu_code
    loop_opt.eliminate_zeros()
../../src/deps/COFFEE/coffee/optimizer.py:163: in eliminate_zeros
    self.nz_syms = zls.reschedule(self.header)
../../src/deps/COFFEE/coffee/scheduler.py:844: in reschedule
    self.exprs.update(elf.fission(stmt, expr_info))
../../src/deps/COFFEE/coffee/scheduler.py:463: in fission
    split, splittable = self.cutter.cut(*splittable)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <coffee.scheduler.CutterSum object at 0x7f2661019650>, node = <coffee.base.Incr object at 0x7f2660e42310>, expr_info = <coffee.expression.MetaExpr object at 0x7f2662468f50>

    def cut(self, node, expr_info):
        left, right = ExpressionFissioner.Cutter.cut(self, node)
        if self._success:
>           index = expr_info.parent.children.index(node)
E           ValueError: <coffee.base.Incr object at 0x7f2660e42310> is not in list

../../src/deps/COFFEE/coffee/scheduler.py:300: ValueError

```

--- a/coffee/cse.py
+++ b/coffee/cse.py
@@ -155,6 +155,10 @@ class CSEUnpicker(object):
             if not temporary.is_ssa:
                 # ... must be written only once
                 return False
+            if not temporary.is_read:
+                # ... must actually be read by some other temporaries (the output
+                # variables are not)
+                return False
             if temporary.is_static_init:
                 # ... its rvalue must not be an array initializer
                 return False

